src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java

```diff
@@ -240,6 +258,49 @@
         });
     }
 
+    @Issue("JENKINS-30326")
+    @Test
+    public void testGlobalBindingWithAuthorization() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                // configure security
+                story.j.jenkins.setSecurityRealm(story.j.createDummySecurityRealm());
+                story.j.jenkins.setAuthorizationStrategy(new FullControlOnceLoggedInAuthorizationStrategy());
+                // create the user.
+                story.j.jenkins.getUser("dummy");
+                
+                // enable the run as user strategy for the AuthorizeProject plugin
+                Map<String, Boolean> strategies = new HashMap<String, Boolean>();
+                strategies.put(story.j.jenkins.getDescriptor(SpecificUsersAuthorizationStrategy.class).getId(), true);
+                QueueItemAuthenticatorConfiguration.get().getAuthenticators().add(new ProjectQueueItemAuthenticator(strategies));
+
+                // blanket whitelist all methods (easier than whitelisting Jenkins.getAuthentication)
+                story.j.jenkins.getExtensionList(Whitelist.class).add(new BlanketWhitelist());
+
+                String credentialsId = "creds";
+                String secret = "s3cr3t";
+                CredentialsProvider.lookupStores(story.j.jenkins).iterator().next().addCredentials(Domain.global(), new StringCredentialsImpl(CredentialsScope.GLOBAL, credentialsId, "sample", Secret.fromString(secret)));
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+
+                p.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  def authentication = Jenkins.getAuthentication()\n"
+                        + "  echo \"running as user: $authentication.principal\"\n"
+                        + "  withCredentials([[$class: 'StringBinding', credentialsId: '" + credentialsId + "', variable: 'SECRET']]) {\n"
+                        + "    echo 'this will fail if the credentials can not be found'\n"
+                        + "  }\n"
+                        + "}", true));
+                // run the job as a specific user
+                p.addProperty(new AuthorizeProjectProperty(new SpecificUsersAuthorizationStrategy("dummy", true)));
+
+                // the build will fail if we can not locate the credentials
+                WorkflowRun b = story.j.assertBuildStatusSuccess(p.scheduleBuild2(0).get());
+                // make sure this was actually run as a user and not system
+                story.j.assertLogContains("running as user: dummy", b);
+            }
```

Comment:
Fair enough but, while you're at it, I'd still add a simple log contains test to make sure it doesn't fail silently. That should already be covered by the tests for `withCredentials`, but it wouldn't hurt adding it here too (one simple line).